lua/autorun/client/cl_cfc_restart_display.lua

```diff
@@ -0,0 +1,64 @@
+local finalTime
+local keyFrame = 0
+local isOpen = false
+local flashState = false
+
+surface.CreateFont( "CFC_CALIBRI_20PX_DEFAULT", {
+    font = "Calibri",
+    extended = false,
+    size = 40,
+    weight = 500
+})
+
+local function timeLeft()
+    return finalTime - SysTime()
+end
+
+concommand.Add( "clean", function()
+    isOpen = false
+end )
+
+local function restartUI()
+    timer.Create( "CFC_RESTART_FLASHTIMER", 1, 0, function()
+        flashState = not flashState
+    end )
+
+    isOpen = true
+    hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+    local delay = net.ReadInt( 16 )
+    local curTime = SysTime()
+    finalTime = curTime + delay
+    hook.Add( "HUDPaint", "CFCDrawRestartAlert", function()
+
+        if isOpen then
+            keyFrame = math.Clamp( keyFrame + 5, 0, ScrW() )
+        else
+            keyFrame = math.Clamp( keyFrame - 10, 0, ScrW() )
+
+            if keyFrame <= 0 then
+                hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+            end
+        end
+
+        surface.SetDrawColor( Color( 36, 54, 72 ) )
+        surface.PlaySound( "ambient/alarms/warningbell1.wav" )
+        surface.DrawRect( 0, 5, keyFrame, 50 )
+    
+        if flashState and timeLeft() <= 30 then surface.SetDrawColor( Color( 255, 255, 89 ) ) elseif not flashState and timeLeft() <= 30 then surface.SetDrawColor( Color( 255, 0, 0 ) ) elseif timeLeft() > 30 then surface.SetDrawColor( 255, 255, 89 ) end
```

Comment:
Maybe something like this? (Assuming you end up going with the table structure)

```lua
CFCRestart.urgencyTime = 30 -- extracting that 30 out from here, should probably be named something better
CFCRestart.defaultDrawColor = Color( 255, 255, 89 )

local drawColor = self.defaultDrawColor

if timeLeft() <= self.urgencyTime then
    if not flashState then
        drawColor = Color( 255, 0, 0 )
    end
end

-- From the wiki: "Providing a Color structure is slower than providing four numbers. You may use Color:Unpack for this."
surface.SetDrawColor( drawColor:Unpack() )
```

Reply:
Applied these changes